tests/UKHO.S100PermitService.Common.UnitTests/Services/ManufacturerKeyServiceTests.cs

```diff
@@ -0,0 +1,127 @@
+
+            _manufacturerKeyService = new ManufacturerKeyService(_fakeManufacturerKeyVault, _fakeCacheProvider, _fakeSecretClient);
+        }
+
+        [Test]
+        public void WhenApplicationStarts_ThenFetchSecretsFromKeyvaultInMemoryCache()
+        {
+            _fakeManufacturerKeyVault.Value.ServiceUri = "https://test.com/";            
+            var secretKey = "mpn";
+            var secretValue = "M_IDmpm";
+
+            A.CallTo(() => _fakeCacheProvider.GetCacheKey(A<string>.Ignored)).Returns(string.Empty);
+
+            A.CallTo(() => _fakeSecretClient.GetPropertiesOfSecrets()).Returns(GetSecretProperties(secretKey));
+
+            A.CallTo(() => _fakeSecretClient.GetSecret(A<string>.Ignored)).Returns(GetSecret(secretKey, secretValue));
+
+            A.CallTo(() => _fakeCacheProvider.SetCacheKey(A<string>.Ignored, A<string>.Ignored, A<TimeSpan>.Ignored)).Returns(secretValue);
+            var result = () => _manufacturerKeyService.CacheManufacturerKeys();
+            result.Should().NotBeNull();           
+        }
+
+        [Test]
+        public void WhenNoSecretsInMemoryCache_ThenThrowException()
+        {
+            _fakeManufacturerKeyVault.Value.ServiceUri = "https://test.com/";
+
+            var result = () => _manufacturerKeyService.CacheManufacturerKeys();
+            result.Should().Throw<PermitServiceException>().WithMessage("No Secrets found in Manufacturer Keyvault");
+
+            A.CallTo(() => _fakeSecretClient.GetPropertiesOfSecrets()).MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        public void WhenValidSecretKeyIsPassed_ThenFetchValueFromMemoryCache()
+        {
+            var secret = SetCacheKeyValue();
+            var secretKey = secret.FirstOrDefault(s => s.Key == "abc").Value;
+
+            A.CallTo(() => _fakeCacheProvider.GetCacheKey(A<string>.Ignored)).Returns(secretKey);
+
+            var result = _manufacturerKeyService.GetManufacturerKeys("abc");
+
+            result.Should().NotBeNull();
+            result.Equals(secretKey);
+        }
+
+        [Test]
+        public void WhenSecretKeyPassedWhichIsNotInMemoryCache_ThenFetchSecretsFromKeyvault()
+        {
+            var secretKey = "mpn";
+            var secretValue = "M_IDmpm";
+
+            A.CallTo(() => _fakeCacheProvider.GetCacheKey(A<string>.Ignored)).Returns(string.Empty);
+
+            A.CallTo(() => _fakeSecretClient.GetSecret(A<string>.Ignored)).Returns(GetSecret(secretKey, secretValue));
+
+            A.CallTo(() => _fakeCacheProvider.SetCacheKey(A<string>.Ignored, A<string>.Ignored, A<TimeSpan>.Ignored)).Returns(secretValue);
+
+            var result = _manufacturerKeyService.GetManufacturerKeys(secretKey);
+
+            result.Should().NotBeNull();
+            result.Should().Be(secretValue);
+        }
+
+        [Test]
+        public void WhenSecretKeyPassedWhichIsNotInMemoryCacheandNotInKeyvault_ThenThrowException()
```

Comment:
Please rename WhenSecretKeyPassedWhichIsNotInMemoryCacheAndNotInKeyVault_ThenThrowException from WhenSecretKeyPassedWhichIsNotInMemoryCacheandNotInKeyvault_ThenThrowException.

Reply:
Removed Unit Test as Exception is handled globally